webapp/src/main/java/com/box/l10n/mojito/quartz/QuartzPollableTaskScheduler.java

```diff
@@ -148,6 +156,12 @@ public <I, O> PollableFuture<O> scheduleJob(QuartzJobInfo<I, O> quartzJobInfo) {
         scheduler.scheduleJob(jobDetail, trigger);
       } else {
         logger.debug("Job already scheduled for key: {}, reschedule", keyName);
+        if (cleanupOnUniqueIdReschedule
+            && dbUtils.isQuartzMysql()
+            && quartzJobInfo.getUniqueId() != null
+            && jobOutputType.equals(Void.class)) {
```

Comment:
would you mind adding a comment saying we only apply the logic to task that have no return type since for others we'd require a way to communicate it was skipped and has no output/is not failed.

Reply:
Done, also added a unit test to verify that a non void output job in the same scenario doesn't get skipped